Afs mi go turrets

Maleclypse opened the pull request:
####Summary
SUMMARY: Mods "Mi-go Turrets"

####Purpose of change
To give purpose to Mi-Go slaves. Replacing PR #35721.

####Describe the solution
I added the the classic Mi-go brain powered turret.

####Describe alternatives you've considered
Not making abominations against man and god. Arming it with the pain beam.

####Testing
TBD in the morning.

####Additional context
They Parrot.

Maleclypse commented:
@Zireael07 For future reference should I delete branches once I merge commits from them or should I leave them around for some reason?

Zireael07 commented:
In 99% of the cases, deleting them after the merge goes through is the way to go. I believe GitHub's online interface even offers you a button to do it easily when the PR is merged. Since the files are on the master repo already you have no practical reason for keeping them (the master repo knows you are a contributor and with web-based interfaces such as GitHub, finding your PRs is a click away).

thethunderhawk commented:
Wait, you can uncraft this thing into human technology? Mi-gos use plutonium cells? They use robotic sensor and identification modules that a human hacker could natively understand and use? That has big lore implications...

Maleclypse commented:
> Wait, you can uncraft this thing into human technology? Mi-gos use plutonium cells? They use robotic modules that a human hacker could natively understand and use? That has big lore implications.
> 
> IMO this thing should be biological, maybe just dropping meat and brains and such. Or else at least it should be made of truly alien technology. Unless the Mi-gos are scavenging this tech from earth...

I used the plutonium cells because we know from the Lore that the high end energy cells that ultratech uses is reverse engineered from Mi-Go tech as are CBMs bio plastic.  I'm iffy about leaving in the robot modules and my plan for the next PR is to replace the ``fluid_preserved_brain`` with a "live brain in a jar", that can be used instead of AI modules for certain robot recipes and will cause the robot to randomly say things like "I have no mouth and yet I scream!"

anothersimulacrum commented on data/mods/Aftershock/afs_speech.json:
```diff
@@ -0,0 +1,44 @@
+  {
+    "type": "speech",
+    "speaker": "afs_mon_migoturret",
+    "sound": "\"Porque?\"",
```
```suggestion
    "sound": "\"Por que?\"",
```
porque is because, por que is why

Maleclypse commented:
@anothersimulacrum Thanks I always mess that up when I haven't looked at other languages for a while.  What do I push to ? upside down at the front?

anothersimulacrum commented on data/mods/Aftershock/afs_speech.json:
```diff
@@ -0,0 +1,44 @@
+  {
+    "type": "speech",
+    "speaker": "afs_mon_migoturret",
+    "sound": "\"Por qué?\"",
```
I was suggesting you add this:
```suggestion
    "sound": "\"¿Por qué?\"",
```

Maleclypse replied:
Thank you!

I-am-Erk commented:
You can do what you like in a mod, but I'm not sure where you got the idea that

> we know from the Lore that the high end energy cells that ultratech uses is reverse engineered from Mi-Go tech as are CBMs bio plastic

Bioplastic for cbms is reverse engineered from migo carapaces, but plutonium fuel cells are dimensional tech built around exotic matter from other dimensions. They have nothing to do with migo.

Maleclypse commented:
I looked back over the timeline document and had thought the first interdimensional traveler was a mi-go but then saw that it’s a unique traveler and that the music-go are encountered first time at -10 years to Cata. I duu I d read that they are scavenging a bunch of earth tech and resources in the design doc section for mi-go. Do you have any preferences of what things would be scavenged and what would be uniquely theirs?

I-am-Erk commented:
They aren't scavenging tech, they're scavenging slaves. Non-biological tech is religious anathema to them.

Maleclypse commented:
https://cddawiki.chezzo.com/cdda_wiki/index.php?title=Lore

Is there an updated Mi-go section somewhere else because this was what I had based my design around. Scavenged tech and research into life forms. No stress if there isn’t.

I-am-Erk commented:
Here, I summarized what I could think of off the top of my head on [the design doc](https://cataclysmdda.org/design-doc/#world) down at the bottom of the World section. I'm running it by Kevin to see if there's anything I missed or misremembered.

thethunderhawk commented:
*another* hivemind!?

I-am-Erk commented:
They aren't in fact a hivemind, but for lore purposes that is something people have hypothesized.

Maleclypse commented:
> Here, I summarized what I could think of off the top of my head on [the design doc](https://cataclysmdda.org/design-doc/#world) down at the bottom of the World section. I'm running it by Kevin to see if there's anything I missed or misremembered.

Thanks @I-am-Erk,  I appreciate your gathering the current data together so I can make something coherent within the lore.  I'm in process of replacing the tech parts with organic equivalents.  I have plans for one specific part to be player usable as is but adding more for flavor and potential future use seemed wise while I was here.  The idea that the player might be able to make some use of the pieces has me keeping this in Aftershock since I want it to be a more sci-fi heavy world.

Maleclypse commented:
> _Another_ hivemind!? Man, these days “Earth Prime” (what do we call our earth?) seems to be the only place in the multiverse with individual, autonomous organisms.
> 
> @Maleclypse I deleted that first comment because I hadn’t realized at first this was mod content. IMO though you just put it a step in the right direction by adding more organic drops, but I saw you kept the humantech robot modules. I’m curious what the lore plans are for Aftershock, and to what extent things like that are just going to be handwaved.

So my original plan was to follow what I thought the lore was and I was using both my own misconceptions and some out of date mi-go lorecraft.  Now that I have a better view of the lore direction, I've made this completely organic tech with a human brain as the CPU.  Currently I am taking Aftershock in two directions.  A high tech fraternity of bionic preppers which I've already added one outpost of to the mod. Secondly, I plan on introducing at least one NPC who is studying the Mi-go and will hopefully teach the PC how to make some utilization of some mi-go tech.  The brain in the jar I plan to use as an alternate AI core and depending on how prototype cyborgs are turned into NPCs I'd love to eventually turn broken cyborg corpses or robot chassis without AI cores into NPCs via Brain in Jar though I'm expecting that would require C++ that isn't hookable through json currently.  I'm open to more suggestions for Aftershock.

Night-Pryanik commented on data/mods/Aftershock/items/afs_bioparts.json:
```diff
@@ -0,0 +1,92 @@
+    "type": "GENERIC",
+    "category": "other",
+    "name": "humming heart",
+    "description": "This heartlike organ has multiple valves and upon it's proper removal from the bioweapons corpse each valve sealed itself. It hums gently in your hands, as if waiting for an appropriate receptacle.",
```
```suggestion
    "description": "This heartlike organ has multiple valves and upon its proper removal from the bioweapons corpse each valve sealed itself.  It hums gently in your hands, as if waiting for an appropriate receptacle.",
```

Night-Pryanik commented on data/mods/Aftershock/items/afs_bioparts.json:
```diff
@@ -0,0 +1,92 @@
+    "type": "GENERIC",
+    "category": "other",
+    "name": "living brain in a jar",
+    "description": "This jar contains a human brain kept alive by Mi-Go technology. It has metal veined tendrils that appear to allow it to connect to various devices.",
```
```suggestion
    "description": "This jar contains a human brain kept alive by mi-go technology.  It has metal veined tendrils that appear to allow it to connect to various devices.",
```

Night-Pryanik commented on data/mods/Aftershock/items/afs_bioparts.json:
```diff
@@ -0,0 +1,92 @@
+    "id": "living_brain_jar",
+    "type": "GENERIC",
+    "category": "other",
+    "name": "living brain in a jar",
```
Plural name needed.

Night-Pryanik commented on data/mods/Aftershock/items/afs_bioparts.json:
```diff
@@ -0,0 +1,92 @@
+    "type": "GENERIC",
+    "category": "other",
+    "name": "sensory cluster",
+    "description": "This lump of flesh has various lobes protruding from it that at intervals will emit toned sounds. Perhaps it works via echolocation. Like all Mi-go bioparts the openings on this piece closed up after it was cut free and it seems to have gone into a form of hibernation with the pulses occurring slower and slower as it conserves energy.",
```
```suggestion
    "description": "This lump of flesh has various lobes protruding from it that at intervals will emit toned sounds.  Perhaps it works via echolocation. Like all mi-go bioparts the openings on this piece closed up after it was cut free and it seems to have gone into a form of hibernation with the pulses occurring slower and slower as it conserves energy.",
```

Night-Pryanik commented on data/mods/Aftershock/items/afs_bioparts.json:
```diff
@@ -0,0 +1,92 @@
+    "id": "bioweapon_chassis",
+    "type": "GENERIC",
+    "category": "other",
+    "name": "bioweapon chassis",
```
Plural name needed.

Night-Pryanik commented on data/mods/Aftershock/items/afs_bioparts.json:
```diff
@@ -0,0 +1,92 @@
+    "type": "GENERIC",
+    "category": "other",
+    "name": "bioweapon chassis",
+    "description": "This cross between a sea anemome and what you imagine a dragon's mouth would look like is the core body of the bioweapon. You cannot imagine what you could make out of this but maybe someone somewhere does.",
```
```suggestion
    "description": "This cross between a sea anemone and what you imagine a dragon's mouth would look like is the core body of the bioweapon.  You cannot imagine what you could make out of this but maybe someone somewhere does.",
```

Night-Pryanik commented on data/mods/Aftershock/items/afs_bioparts.json:
```diff
@@ -0,0 +1,92 @@
+    "looks_like": "ar15",
+    "type": "GUN",
+    "name": "vibrating bioblaster",
+    "description": "You ripped this from a Mi-go abomination. You have no clue how or if it can be reloaded. ",
```
```suggestion
    "description": "You ripped this from a mi-go abomination.  You have no clue how or if it can be reloaded. ",
```

Night-Pryanik commented on data/mods/Aftershock/items/afs_bioparts.json:
```diff
@@ -0,0 +1,92 @@
+    "price": 100,
+    "to_hit": -1,
+    "bashing": 5,
+    "material": "resin",
```
```suggestion
    "material": "alien_resin",
```

Night-Pryanik commented on data/mods/Aftershock/items/afs_bioparts.json:
```diff
@@ -0,0 +1,92 @@
+    "price": 100,
+    "to_hit": -1,
+    "bashing": 1,
+    "material": "resin",
```
```suggestion
    "material": "alien_resin",
```

Night-Pryanik commented on data/mods/Aftershock/items/afs_bioparts.json:
```diff
@@ -0,0 +1,92 @@
+    "price": 100,
+    "to_hit": -3,
+    "bashing": 7,
+    "material": "resin",
```
```suggestion
    "material": "alien_resin",
```

Night-Pryanik commented on data/mods/Aftershock/items/afs_inactiverobot.json:
```diff
@@ -0,0 +1,24 @@
+    "type": "TOOL",
+    "name": "inactive brain blaster",
+    "description": "This is a horrifying monstrosity.  Using this item is cannibalism in all but taste and involves turning it on and placing it on the ground, where it will attach itself.  If reprogrammed and rewired successfully the turret will identify you as a friendly, and attack all enemies with its revolving laser cannons.",
+    "weight": "1450 g",
```
```suggestion
    "weight": "40750 g",
```

Night-Pryanik commented on data/mods/Aftershock/items/afs_inactiverobot.json:
```diff
@@ -0,0 +1,24 @@
+    "name": "inactive brain blaster",
+    "description": "This is a horrifying monstrosity.  Using this item is cannibalism in all but taste and involves turning it on and placing it on the ground, where it will attach itself.  If reprogrammed and rewired successfully the turret will identify you as a friendly, and attack all enemies with its revolving laser cannons.",
+    "weight": "1450 g",
+    "volume": "3 L",
```
```suggestion
    "volume": "30 L",
```

Night-Pryanik commented on data/mods/Aftershock/items/afs_inactiverobot.json:
```diff
@@ -0,0 +1,24 @@
+      "monster_id": "afs_mon_migoturret",
+      "difficulty": 6,
+      "moves": 100,
+      "skill1": "cooking",
```
Cooking? Is this right?

Maleclypse replied:
I'm kinda torn on this one and open to suggestions, maybe persuasion, maybe first aid? Cooking has chemistry in it which I think ties into biotech more than computers, electronics or mechanics.

ZhilkinSerg replied:
I think it is okay for now as cooking also includes chemistry. It could be updated later when we have more skills.

Night-Pryanik commented on data/mods/Aftershock/mobs/afs_robots.json:
```diff
@@ -0,0 +1,38 @@
+    "id": "afs_mon_migoturret",
+    "type": "MONSTER",
+    "name": "brain blaster",
+    "description": "The Mi-Go have set up some defences here.  This horrifying techo-organic amalgamation appears to use brains in jars as their operating system. On the plus side it looks like it can be headshot.",
```
```suggestion
    "description": "The mi-go have set up some defenses here.  This horrifying techo-organic amalgamation appears to use brains in jars as their operating system.  On the plus side it looks like it can be headshot.",
```

Night-Pryanik commented on data/mods/Aftershock/mobs/afs_robots.json:
```diff
@@ -0,0 +1,38 @@
+        "gun_type": "laser_cannon",
+        "fake_skills": [ [ "gun", 2 ], [ "rifle", 3 ] ],
+        "ranges": [ [ 0, 20, "DEFAULT" ] ],
+        "require_sunlight": false
```
```suggestion
```

Redundant, it defaults to false anyway.
